addons/account_payment/views/account_portal_templates.xml

```diff
@@ -1,13 +1,13 @@
 <odoo>
     <template id="portal_my_invoices_payment" name="Payment on My Invoices" inherit_id="account.portal_my_invoices">
-        <xpath expr="//t[@t-call='portal.portal_table']/thead/tr/th[last()]" position="before">
+        <xpath expr="//t[@t-call='portal.portal_table']/thead/tr/th[last() - 1]" position="before">
             <th class="text-center">Status</th>
         </xpath>
         <xpath expr="//t[@t-foreach='invoices']/tr/td[last()]" position="before">
             <td class="text-center">
                 <t t-set="tx_ids" t-value="invoice.transaction_ids.filtered(lambda tx: tx.state in ('pending', 'authorized', 'done'))"/>
                 <t t-set="pending_manual_txs" t-value="tx_ids.filtered(lambda tx: tx.state == 'pending' and tx.provider in ('none', 'transfer'))"/>
-                <a t-if="invoice.state == 'posted' and invoice.payment_state in ('not_paid', 'partial') and invoice.amount_total and invoice.move_type == 'out_invoice' and (pending_manual_txs or not tx_ids or invoice.amount_residual)"
+                <a t-if="last_tx.state != 'done' and invoice.state == 'posted' and invoice.payment_state in ('not_paid', 'partial') and invoice.amount_total and invoice.move_type == 'out_invoice' and (pending_manual_txs or not tx_ids or invoice.amount_residual)"
```

Comment:
I'm working on a PR in master that will fix that (and other bugs)
See this commit: https://github.com/odoo/odoo/pull/78843/commits/9a6bd5b9980bca27d9df44712a8ad00c6620ac7a

I've tested those changes and it seems to be working in 15.0 too.

If the last tx is `done`, it doesn't mean that the invoice is fully paid.